Hw3 page objects

andrei-mikhailovskii opened the pull request.

andrei-mikhailovskii committed:
create first page objects, start modifying test classes

andrei-mikhailovskii committed:
add and modify page objects: IndexPageNavBar, IndexPageBenefitsSections, IndexPageIFrame
modify AbstractChromeTest, ExerciseOneTest

andrei-mikhailovskii committed:
modified locators in ExerciseTwoTest.java, IndexPageNavBar.java

andrei-mikhailovskii committed:
finished page objects fro the first test, started modifying ExerciseTwoTest.java

andrei-mikhailovskii committed:
modified ExerciseTwoTest.java, filled DifferentElementsPage page object, add DifferentElementsPageMainContainer page object

andrei-mikhailovskii committed:
continue filling DifferentElementsPageMainContainer and modifying tests respectively

andrei-mikhailovskii committed:
pre-finished page objects and test classes, added Maven profile for hw3

andrei-mikhailovskii committed:
added config.properties file, PropertiesExtractor class, modified ExerciseOneTest, ExterciseTwoTest (provided userName and userPassword from the config.properties)

andrei-mikhailovskii committed:
modified files from hw2 according to the comments on GitHub

andrei-mikhailovskii committed:
made DifferentElementsPage page object void

andrei-mikhailovskii committed:
add allure directory to gitignore
add Allure and AspectJ dependencies to POM.xml
fill ExerciseOneTest in hw3/ex1 directory
fill ExerciseTwoTest in hw3/ex2 directory
delete assertions from DifferentElementsPageMainContainer, IndexPage, IndexPageIframe, IndexPageLeftSection, IndexPageNavBar, IndexPageMainContainer
add HeaderMenu, RightSideBar object pages
add allure.properties

andrei-mikhailovskii committed:
delete HeaderMenu page object as unnscessary (HeaderNavBar page component has the same code)
refactor IndexPageNaveBar - rename to HeaderNavBar, make it page component
refactor IndexPageLeftSection - rename to LeftNavBar, make it page component

andrei-mikhailovskii committed:
add PageFactory for HeaderNavBar and LeftNavBar

knadonenkoepam commented on src/test/java/com/epam/tc/hw3/page/IndexPageMainContainer.java:
```diff
@@ -0,0 +1,86 @@
+    @FindBy(className = "benefit-txt")
+    private List<WebElement> listOfTexts;
+
+    @FindBy(xpath = "//div[normalize-space(span) = 'To include good practicesand ideas from successfulEPAM project']")
```
the same thing which you did in EX2, you don't need these elements, just one and verify that elements in list has the text

andrei-mikhailovskii replied:
removed excessive FindBy annotations with texts, and now a list of WebElements is used in "for" loop with getText method

knadonenkoepam commented on src/test/java/com/epam/tc/hw3/page/IndexPage.java:
```diff
@@ -0,0 +1,76 @@
+
+    public IndexPage loginUser(WebDriver driver, String userName, String userPassword) {
+        if (notLoggedInUserNames.size() != 0) {
+            new WebDriverWait(driver, WAIT_TIMEOUT_SEC)
```
It is better to create a dedicated class which would be responsible for WebDriverWait

andrei-mikhailovskii replied:
added WebDriverWaitClass with respective methods

knadonenkoepam commented on src/test/java/com/epam/tc/hw3/ex2/ExerciseTwoTest.java:
```diff
@@ -0,0 +1,64 @@
+public class ExerciseTwoTest extends AbstractChromeTest {
+
+    private String correctPageTitle = "Home Page";
+    private String correctUserName = "ROMAN IOVLEV";
```
Username also could be moved and taken from resources

andrei-mikhailovskii replied:
Moved Username to the config.properties file

andrei-mikhailovskii committed:
add correctUsername to config.properties and use it in PropertiesExtractor class
add WebDriverWaitClass and refactor other classes which use it
in IndexPageMainContainer class, replace getTextPractise method with numberOfCorrectTexts one

andrei-mikhailovskii committed:
rename webdriver in AbstractChromeTest

andrei-mikhailovskii committed:
rename webdriver in AbstractChromeTest

andrei-mikhailovskii committed:
Removed hw2 files from pull request

andrei-mikhailovskii committed:
Merge branch 'main' into hw3-page-objects
